container/container.go

```diff
@@ -373,35 +373,46 @@ func (container *Container) GetExecIDs() []string {
 	return container.ExecCommands.List()
 }
 
-// Attach connects to the container's TTY, delegating to standard
-// streams or websockets depending on the configuration.
-func (container *Container) Attach(stdin io.ReadCloser, stdout io.Writer, stderr io.Writer, keys []byte) chan error {
-	ctx := container.InitAttachContext()
-	return AttachStreams(ctx, container.StreamConfig, container.Config.OpenStdin, container.Config.StdinOnce, container.Config.Tty, stdin, stdout, stderr, keys)
-}
-
 // AttachStreams connects streams to a TTY.
 // Used by exec too. Should this move somewhere else?
-func AttachStreams(ctx context.Context, streamConfig *stream.Config, openStdin, stdinOnce, tty bool, stdin io.ReadCloser, stdout io.Writer, stderr io.Writer, keys []byte) chan error {
+func AttachStreams(streamConfig *stream.Config, openStdin, useStdin, useStdout, useStderr bool) (io.WriteCloser, io.ReadCloser, io.ReadCloser) {
 	var (
 		cStdout, cStderr io.ReadCloser
 		cStdin           io.WriteCloser
-		wg               sync.WaitGroup
-		errors           = make(chan error, 3)
 	)
 
-	if stdin != nil && openStdin {
+	if useStdin && openStdin {
 		cStdin = streamConfig.StdinPipe()
+	}
+
+	if useStdout {
+		cStdout = streamConfig.StdoutPipe()
+	}
+
+	if useStderr {
+		cStderr = streamConfig.StderrPipe()
+	}
+
+	return cStdin, cStdout, cStderr
+}
+
+// CopyStreams starts goroutines to copy data in and out to/from the container
+// Used by exec too. Should this move somewhere else?
+func CopyStreams(ctx context.Context, streamConfig *stream.Config, openStdin, stdinOnce, tty bool, stdin io.ReadCloser, stdout io.Writer, stderr io.Writer, cStdin io.WriteCloser, cStdout io.ReadCloser, cStderr io.ReadCloser, keys []byte) chan error {
```

Comment:
* `streamConfig` seems unused
* having a dozen of arguments does not seem a good design
* Since this function is not used in the `container` package, you can move this to `daemon` package. Maybe it is even ok to move to under `pkg/something`.